src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -606,18 +607,37 @@ def qualification(self):
         """Return the qualification type of this reference."""
         return "qualified" if self.is_qualified() else "unqualified"
 
-    def extract_reference(self, level: int) -> Optional[ObjectReferencePart]:
-        """Extract a reference of a given level.
+    class ReferencePart(Enum):
+        """Labels for the "levels" of a reference.
+
+        Note: Since SQLFluff does not have access to database catalog
+        information, interpreting references will often be ambiguous.
+        Typical example: The first part *may* refer to a schema, but that is
+        almost always optional if referring to an object in some default or
+        currently "active" schema. For this reason, use of this enum is optional
+        and intended mainly to clarify the intent of the code -- no guarantees!
+        Additionally, the terminology may vary by dialect, e.g. in BigQuery,
+        "project" would be a more accurate term than "schema".
+        """
+
+        OBJECT = 1
+        TABLE = 2
+        SCHEMA = 3
+
+    def extract_possible_references(self, level: int) -> List[ObjectReferencePart]:
```

Comment:
can't that be `level: ReferencePart` now?

Reply:
I want to support both integers or enums here, because of the ambiguity I commented on in the enum. You make a good point, though -- I have updated the type hint to specify that either is allowed and to handle both correctly. 👍

In BigQuery, for example, a reference could have literally dozens of parts (for example, a highly nested structure). The enum is useful for the 3 typical or "canonical" parts. Currently, the only code that calls this function happens to use the enum, but I could easily see other future users of this class & function wanting to just specify a number, e.g. if they were looping over all the parts or something. I could see revisiting this code in the future. It's challenging to find the right balance of clarity and flexibility here, because of the limited number of "data points" (i.e. code that calls this function).